src/Http/Http.Extensions/gen/StaticRouteHandlerModel/Emitters/EndpointEmitter.cs

```diff
@@ -61,13 +61,15 @@ public static void EmitRouteOrQueryResolver(this Endpoint endpoint, CodeWriter c
         }
     }
 
-    public static void EmitJsonBodyOrServicePreparation(this Endpoint endpoint, CodeWriter codeWriter)
+    public static void EmitJsonBodyOrServiceResolver(this Endpoint endpoint, CodeWriter codeWriter)
     {
         foreach (var parameter in endpoint.Parameters)
         {
             if (parameter.Source == EndpointParameterSource.JsonBodyOrService)
             {
                 codeWriter.WriteLine("var serviceProviderIsService = options?.ServiceProvider?.GetService<IServiceProviderIsService>();");
+                codeWriter.Write($@"var {parameter.Name}_JsonBodyOrServiceResolver = ");
```

Comment:
This includes the parameter name in the variable name. What is the parameter name here? Is it the C# method parameter name or the route parameter name? Need to be careful about using the route name because it could contain invalid C# name characters.

Reply:
I looked into this and I believe the parameter name here can be overridden by something like `[FromRoute(Name = "1")]`. The generated source wouldn't compile because a variable can't start with a number.

Consider using the parameter index instead, e.g. `param_1__JsonBodyOrServiceResolver`

Reply:
This should be based on the parameter name even if it can be overridden. That should just change the lookup key, not the variable name.